### ISSUE TITLE
[no-empty-interface] Allow an empty interface to extend a type alias

### ISSUE THREAD

**Issue by laurence-myers** [typescript-eslint/typescript-eslint#3233]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```javascript
{
  parser: '@typescript-eslint/parser',
  parserOptions: {
    tsconfigRootDir: __dirname,
    project: [
      './tsconfig.json',
    ],
    sourceType: 'module',
  },
  plugins: ['@typescript-eslint/eslint-plugin'],
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:@typescript-eslint/recommended-requiring-type-checking',
  ],
  root: true,
  env: {
    node: true,
    jest: true,
  },
}
```

```TS
type Foo {
  bar: string;
}

interface Baz extends Foo {}
```

**Expected Result**

No lint warning

**Actual Result**

`ESLint: An interface declaring no members is equivalent to its supertype.(@typescript-eslint/no-empty-interface)`

**Additional Info**

TypeScript allows defining an interface that extends from the type alias, _when_ the type alias can be reduced to a static type by TSC. (Type unions etc won't work)

This is useful for giving a short name for a type, for improved error messages or IDE integration.

It would be nice to have an option to toggle the warning for this scenario.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.0` |
| `@typescript-eslint/parser`        | `4.14.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `^7.18.0` |
| `node`                             | `14.15.4` |

**Comment by bradzacher** [comment]
this rule operates solely on the AST - it does not use type information - so this is not possible.
It's also impossible to tell developer intent.
"did you mean to use extends, or did you do it accidentally?"

Use a disable comment for these cases.
